### PR TITLE
decouple Create and SetCurrent for app and component

### PR DESCRIPTION
**Pull request by kadel**
fixes https://github.com/redhat-developer/ocdev/issues/180

Also, this will make it easier to write unit tests when #232 is done

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -201,6 +196,36 @@ func (c *ConfigInfo) SetActiveApplication(application string, project string) er
...
+
+	for i, app := range c.ActiveApplications {
+		// if application exists set is as Active
+		if app.Name == application && app.Project == project {
+			c.ActiveApplications[i].Active = true
+			return fmt.Errorf("unable to set add %s application, it already exists in config file", application)
+		}
+	}
+
+	// if application doesn't exists, add it as Active
```
Comment (containscafeine) [inline pkg/config/config.go]: should this comment be here, we are not setting the application as active

File: pkg/config/config.go [hunk 2]
```diff
@@ -201,6 +196,36 @@ func (c *ConfigInfo) SetActiveApplication(application string, project string) er
...
 }
 
+// AddApplication add  new application to the config file
+// Newly create application is NOT going to be se as Active.
+func (c *ConfigInfo) AddApplication(application string, project string) error {
+	if c.ActiveApplications == nil {
+		c.ActiveApplications = []ApplicationInfo{}
+	}
+
+	for i, app := range c.ActiveApplications {
```
Comment (containscafeine) [inline pkg/config/config.go]: this loop sets the application as active. we already have `SetActiveApplication()`, should we have this here?

Comment (kadel) [reply pkg/config/config.go]: Thanks for catching this.
Setting application as active shouldn't be there. This loop is there only to error out if the application already exists.

fixed

File: pkg/config/config.go (hunk 1 above)
Comment (kadel) [reply pkg/config/config.go]: correct, I  forgot to update that comment.

fixed

**Pull request merged**